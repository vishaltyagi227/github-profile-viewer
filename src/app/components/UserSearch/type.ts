export interface ISearch {
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}
