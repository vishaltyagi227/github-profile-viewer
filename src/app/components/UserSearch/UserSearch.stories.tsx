import React from 'react';
import { Story, Meta } from '@storybook/react';
import UserSearch from '@components/UserSearch';
import { ISearch } from '@components/UserSearch/type';

export default {
  title: 'Components/UserSearch',
  component: UserSearch,
  argTypes: {
    handleChange: { action: 'change' },
  },
} as Meta;

const Template: Story<ISearch> = ({ ...args }) => <UserSearch {...args} />;

export const Basic = Template.bind({});
