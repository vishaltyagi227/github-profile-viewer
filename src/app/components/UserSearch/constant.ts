const SEARCH_CONFIG = {
  type: 'search',
  name: 'search',
  placeholder: 'Search User',
};

export default SEARCH_CONFIG;
