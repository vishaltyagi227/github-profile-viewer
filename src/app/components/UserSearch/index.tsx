import React from 'react';
import { TextField, InputAdornment } from '@mui/material';
import { Search } from '@mui/icons-material';

import { POSITION, AUTO_COMPLETE } from '@constant/index';
import useStyles from './style';
import { ISearch } from './type';
import SEARCH_CONFIG from './constant';

const UserSearch = ({ handleChange }: ISearch) => {
  const classes = useStyles();
  return (
    <TextField
      placeholder={SEARCH_CONFIG.placeholder}
      type={SEARCH_CONFIG.type}
      name={SEARCH_CONFIG.name}
      onChange={handleChange}
      className={classes.searchInput}
      autoComplete={AUTO_COMPLETE.off}
      InputProps={{
        startAdornment: (
          <InputAdornment position={POSITION.start}>
            <Search />
          </InputAdornment>
        ),
      }}
    />
  );
};

export default UserSearch;
