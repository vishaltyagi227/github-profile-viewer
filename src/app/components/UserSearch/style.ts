import { Theme } from '@mui/material/styles';
import { createStyles, makeStyles } from '@mui/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    searchWrapper: {
      padding: theme.spacing(5),
    },
    searchInput: {
      marginBlock: theme.spacing(3),
      width: '100%',
    },
  })
);

export default useStyles;
