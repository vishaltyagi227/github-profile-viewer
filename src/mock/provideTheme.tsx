import React, { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material/styles';
import theme from '@theme/index';

export default (ui: ReactElement): ReactElement => (
  <ThemeProvider theme={theme}>{ui}</ThemeProvider>
);
