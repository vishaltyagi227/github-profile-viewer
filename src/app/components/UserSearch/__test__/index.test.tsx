import React from 'react';
import { cleanup, render, RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';
import provideTheme from '@mock/provideTheme';
import UserSearch from '../index';

describe('<UserSearch />', () => {
  let container: RenderResult;

  afterEach(cleanup);

  it('Should render correctly', () => {
    container = render(provideTheme(<UserSearch handleChange={jest.fn()} />));
    expect(container.getByPlaceholderText('Search User')).toBeInTheDocument();
  });
});

describe('<UserSearch /> Snapshot', () => {
  afterEach(cleanup);

  it('Should match with Snapshot', () => {
    const tree = renderer
      .create(provideTheme(<UserSearch handleChange={jest.fn()} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
