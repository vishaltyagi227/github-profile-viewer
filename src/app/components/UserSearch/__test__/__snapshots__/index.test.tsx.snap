// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserSearch /> Snapshot Should match with Snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-searchInput-4 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <input
      aria-invalid={false}
      autoComplete="off"
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedStart css-qyuz5a-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      name="search"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search User"
      required={false}
      type="search"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1a1ihhj"
      >
        <span
          className="notranslate"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
